src/instrumentation/page.ts

```diff
@@ -0,0 +1,85 @@
+import { ReadableSpan } from '@opentelemetry/sdk-trace-base'
+import { Initialiser, setConfig } from '../config'
+import { exportSpans, proxyExecutionContext } from './common'
+import { Exception, SpanKind, SpanOptions, SpanStatusCode, context as api_context, trace } from '@opentelemetry/api'
+import { wrap } from '../wrap'
+import {
+	gatherIncomingCfAttributes,
+	gatherRequestAttributes,
+	gatherResponseAttributes,
+	getParentContextFromRequest,
+} from './fetch'
+
+type PageHandlerArgs = Parameters<PagesFunction>
+
+let cold_start = true
+export function executePageHandler(pagesFn: PagesFunction, [request]: PageHandlerArgs): Promise<Response> {
+	const spanContext = getParentContextFromRequest(request.request)
+
+	const tracer = trace.getTracer('pagesHandler')
+	const attributes = {
+		['faas.trigger']: 'http',
+		['faas.coldstart']: cold_start,
+		['faas.invocation_id']: request.request.headers.get('cf-ray') ?? undefined,
+	}
+	cold_start = false
+	Object.assign(attributes, gatherRequestAttributes(request.request))
+	Object.assign(attributes, gatherIncomingCfAttributes(request.request))
```

Comment:
Could you also assign version metadata, like in https://github.com/evanderkoogh/otel-cf-workers/blob/3bdda1e1774289b4fdfcb3fc9c11ac5b65a9ffca/src/instrumentation/fetch.ts#L144

I know that version metadata bindings can't be added to Pages projects (looking at you @jahands), and script versions work very differently, but hopefully this will change as they converge more.